index.js

```diff
@@ -186,12 +186,9 @@ var publicAgent = {
           }
         }
         if (err) {
-          // Fatal error. Disable the agent.
           logger.error('Unable to acquire the project number from metadata ' +
             'service. Please provide a valid project number as an env. ' +
-            'variable, or through config.projectId passed to start(). ' +
-            'Disabling trace agent. ' + err);
-          publicAgent.stop();
+            'variable, or through config.projectId passed to start(). ' + err);
```

Comment:
I've moved the responsibility of handling metadata-based project ID errors to `trace-writer.js`:

https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/pull/378/commits/b7822a1a820172328f3a4ee9bfdc77aa98405a25#diff-d2b33b8013e140d6fa2ffc2d03dcf77a